### ISSUE TITLE
 "Qty Uses Decimals" doesn't work for simple product

### ISSUE THREAD

**Issue by smoskaluk** [magento/inventory#774]
### Preconditions

1. Magento MSI
2. Simple product created

### Steps to reproduce

1. In "Advanced Inventory" enable "Qty Use Decimals" for your simple product 
2. Go to home page and open simple product page
3. Try to add to cart simple product with qty 1.23

### Expected result

1. Product should be added to the cart

### Actual result

1. Error message: The requested qty is not a valid increment
![image](https://user-images.githubusercontent.com/27818516/37964452-3d7df802-31ca-11e8-984e-b2b5513e6c62.png)

**Comment by OlgaBurtyka** [comment]
I see some other error
![screenshot_20180405_140551](https://user-images.githubusercontent.com/4771407/38362720-8b310cd0-38da-11e8-99ae-77c7669d3d40.png)

**Comment by maghamed** [comment]
We have next code
```
        /**
         * Check quantity type
         */
        $result->setItemIsQtyDecimal($stockItem->getIsQtyDecimal());
        if (!$stockItem->getIsQtyDecimal()) {
            $result->setHasQtyOptionUpdate(true);
            $qty = intval($qty);
            /**
             * Adding stock data to quote item
             */
            $result->setItemQty($qty);
            $qty = $this->getNumber($qty);
            $origQty = intval($origQty);
            $result->setOrigQty($origQty);
        }
```
In legacy implementation
`\Magento\CatalogInventory\Model\StockStateProvider::checkQuoteItemQty`

This case is not covered in MSI

**Comment by OlgaBurtyka** [comment]
I did some changes in commit, which allow save decimal qty_increment in configuration. But it's not all. There in  many places exist issue regarding decimal. For example on admin in product page we can save decimal increment, but if one edit product page after re-save we will 0 in field(qty_increment);
![export 7](https://user-images.githubusercontent.com/4771407/38769103-26902c2e-4006-11e8-986d-8fd99a5636ae.png)
![export 8](https://user-images.githubusercontent.com/4771407/38769104-26b1d3b0-4006-11e8-9242-93dffd54bcbd.png)
![export 9](https://user-images.githubusercontent.com/4771407/38769105-26e552da-4006-11e8-8a69-ab04184ea59f.png)
![export 10](https://user-images.githubusercontent.com/4771407/38769106-27025b46-4006-11e8-8279-2d340853f52f.png)

**Comment by phoenix128** [comment]
Fixed in core branch due to code freeze: https://github.com/magento/magento2/pull/14806

**Comment by naydav** [comment]
Closed due to magento#14806
